core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSailConnection.java

```diff
@@ -866,26 +890,63 @@ public ValidationReport revalidate() {
 		return new ShaclSailValidationException(validate).getValidationReport();
 	}
 
+	Settings getTransactionSettings() {
+		return transactionSettings;
+	}
+
 	public static class Settings {
 
-		private ShaclSail.TransactionSettings.ValidationApproach validationApproach = ShaclSail.TransactionSettings.ValidationApproach.Auto;
-		private boolean cacheSelectedNodes = false;
+		private ShaclSail.TransactionSettings.ValidationApproach validationApproach;
+		private Boolean cacheSelectedNodes;
+		private Boolean parallelValidation;
 
-		public Settings(boolean cacheSelectNodes, boolean validationEnabled) {
+		public Settings() {
+		}
+
+		public Settings(boolean cacheSelectNodes, boolean validationEnabled, boolean parallelValidation) {
 			this.cacheSelectedNodes = cacheSelectNodes;
 			if (!validationEnabled) {
 				validationApproach = ShaclSail.TransactionSettings.ValidationApproach.Disabled;
+			} else {
+				this.validationApproach = ShaclSail.TransactionSettings.ValidationApproach.Auto;
 			}
+			this.parallelValidation = parallelValidation;
 		}
 
 		public ShaclSail.TransactionSettings.ValidationApproach getValidationApproach() {
 			return validationApproach;
 		}
 
 		public boolean isCacheSelectNodes() {
-			return cacheSelectedNodes && validationApproach != ShaclSail.TransactionSettings.ValidationApproach.Bulk;
+			return cacheSelectedNodes;
 		}
 
+		public boolean isParallelValidation() {
```

Comment:
This shouldn't be public I think?

Reply:
The entire Settings class shouldn't be public, but it's public already so a little late now. I will make the applySettings method not public.